File: backend/src/main/java/io/github/artemy/osipov/shop/service/present/CreatePresentCommand.java
package io.github.artemy.osipov.shop.service.present;

import lombok.Builder;
import lombok.Value;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

@Value
@Builder
public class CreatePresentCommand {
    UUID id;
    String name;
    BigDecimal price;
    List<PresentItem> items;

    @Value
    @Builder
    public static class PresentItem {
        UUID candyId;
        Integer count;
    }
}

File: backend/src/main/kotlin/io/github/artemy/osipov/shop/service/present/CreatePresentCommand.kt
package io.github.artemy.osipov.shop.service.present

import java.util.UUID
import java.math.BigDecimal

class CreatePresentCommand(
    val id: UUID,
    val name: String,
    val price: BigDecimal,
    val items: List<PresentItem>
) {
    data class PresentItem(
        val candyId: UUID,
        val count: Int
    )
}

